src/internal/packager/images/pull.go

```diff
@@ -308,6 +311,20 @@ func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Im
 			return nil, false, fmt.Errorf("failed to load image from docker daemon: %w", err)
 		}
 	} else {
+		var idx v1.IndexManifest
+		if err := json.Unmarshal(manifest, &idx); err != nil {
+			return nil, false, lang.ErrUnsupportedImageType
+		}
+
+		if strings.Contains(src, "@") && (idx.MediaType == ctypes.OCIImageIndex || idx.MediaType == ctypes.DockerManifestList) {
+			imageOptions := "please select one of the images below based on your platform"
```

Comment:
Two things: 
I think we should be testing the output of `imageOptions` in our tests, but also I think this may read a little awkward without any line breaks.

Reply:
Yeah kinda the middle ground I've gone for is checking that one of the listed images in the image index is in the stderr in the e2e test. So for one of the tests I check that ghcr.io/defenseunicorns/zarf/agent:v0.32.6@sha256:b3fabdc7d4ecd0f396016ef78da19002c39e3ace352ea0ae4baa2ce9d5958376 is in stderr which is the arm64 image

Agree it's not perfect though and we see things like this in a lot of our e2e tests. Trying to balance readability / not locking tests to messages. Open to checking against more of the output or other suggestions.

Reply:
I agree that `imageOptions` should be checked for in the test. There is currently a bug because `"please select one of the images below based on your platform"` isn't being printed to stderr, and testing for this output would have caught it quickly.

Reply:
>  There is currently a bug because "please select one of the images below based on your platform" isn't being printed to stderr, and testing for this output would have caught it quickly.

Scratch this part. I see it is showing up in the output, but testing for it with our current setup might be weird. But I do think it would be good to test for the output